proofs/proof_global.ml

```diff
@@ -336,10 +336,10 @@ let close_proof ~keep_body_ucst_separate ?feedback_id ~now
   (* Because of dependent subgoals at the beginning of proofs, we could
      have existential variables in the initial types of goals, we need to
      normalise them for the kernel. *)
-  let subst_evar k =
-    Proof.in_proof proof (fun m -> Evd.existential_opt_value m k) in
-  let nf = Universes.nf_evars_and_universes_opt_subst subst_evar
-    (Evd.evar_universe_context_subst universes) in
+  let nf = Proof.in_proof proof (fun sigma ->
+      let sigma = Evd.set_universe_context sigma universes in
+      Evarutil.nf_evars_universes sigma)
+  in
```

Comment:
I'm not sure if this is cleaner with nf_evars_and_universes_opt_subst or using Evd.set_universe_context, but keeping around a separate implementation of EConstr.to_constr is a bit weird.